integration/pushed_authorize_code_grant_test.go

```diff
@@ -0,0 +1,202 @@
+			setup: func() {
+				oauthClient = newOAuth2Client(ts)
+				state = "12345678901234567890"
+			},
+			check: func(t *testing.T, r *http.Response) {
+				var b fosite.AccessRequest
+				b.Client = new(fosite.DefaultClient)
+				b.Session = new(defaultSession)
+				require.NoError(t, json.NewDecoder(r.Body).Decode(&b))
+				assert.EqualValues(t, fosite.Arguments{"https://www.ory.sh/api"}, b.RequestedAudience)
+				assert.EqualValues(t, fosite.Arguments{"https://www.ory.sh/api"}, b.GrantedAudience)
+				assert.EqualValues(t, "foo-sub", b.Session.(*defaultSession).Subject)
+			},
+			parStatusCode:  http.StatusCreated,
+			authStatusCode: http.StatusOK,
+		},
+		{
+			description: "should pass",
+			setup: func() {
+				oauthClient = newOAuth2Client(ts)
+				state = "12345678901234567890"
+			},
+			parStatusCode:  http.StatusCreated,
+			authStatusCode: http.StatusOK,
+		},
+	} {
+		t.Run(fmt.Sprintf("case=%d/description=%s", k, c.description), func(t *testing.T) {
+			c.setup()
+
+			// build request from the OAuth client
+			data := url.Values{}
+			data.Set("client_id", oauthClient.ClientID)
+			data.Set("client_secret", oauthClient.ClientSecret)
+			data.Set("response_type", "code")
+			data.Set("state", state)
+			data.Set("scope", strings.Join(oauthClient.Scopes, " "))
+			data.Set("redirect_uri", oauthClient.RedirectURL)
+			for k, v := range c.params {
+				data.Set(k, v)
+			}
+
+			req, err := http.NewRequest("POST", ts.URL+"/par", strings.NewReader(data.Encode()))
+			require.NoError(t, err)
+
+			req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+			resp, err := http.DefaultClient.Do(req)
+
+			require.NoError(t, err)
+
+			body, err := checkStatusAndGetBody(t, resp, c.parStatusCode)
+			require.NoError(t, err, "Unable to get body after PAR. Err=%v", err)
+
+			if resp.StatusCode != http.StatusCreated {
+				return
+			}
+
+			m := map[string]interface{}{}
+			err = json.Unmarshal(body, &m)
+
+			assert.NoError(t, err, "Error occurred when unamrshaling the body: %v", err)
+			assert.Condition(t, func() bool { return m["request_uri"].(string) != "" }, "request_uri is empty")
+			assert.Condition(t, func() bool {
+				return strings.HasPrefix(m["request_uri"].(string), "urn:ietf:params:oauth:request_uri:")
+			}, "PAR Prefix is incorrect: %v", m["request_uri"].(string))
+			assert.EqualValues(t, 300, int(m["expires_in"].(float64)), "Invalid expires_in value=%v", m["expires_in"])
+
+			// call authorize
+			data = url.Values{}
+			data.Set("client_id", oauthClient.ClientID)
+			data.Set("request_uri", m["request_uri"].(string))
+			req, err = http.NewRequest("POST", ts.URL+"/auth", strings.NewReader(data.Encode()))
+			require.NoError(t, err)
+
+			req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+			resp, err = http.DefaultClient.Do(req)
+			require.NoError(t, err)
+			require.Equal(t, c.authStatusCode, resp.StatusCode)
+			if resp.StatusCode != http.StatusOK {
+				return
```

Comment:
Shouldn't this fail the test or do some other assertion besides the status code check?

Reply:
The failure occurs before this call when we assert the HTTP status code.

Reply:
It does attempt to exchange the code for the token and that would fail if the code isn't successfully returned, but I see your point. Adding a check to verify the code is non-empty.

Reply:
This is the last point - I couldn't find whether you made changes to address this comment. Is it still open or resolved?

Reply:
I see now, you added the check below

Reply:
Right after this if check, there's a check on `code` being non-empty. The validation of the code itself would happen when the request is made to the token endpoint.